src/web3/Client.ts

```diff
@@ -26,15 +33,30 @@ export class Client implements IClient {
    */
   public constructor(
     private clientConfig: IClientConfig,
-    baseAccount?: IAccount,
+    baseAccount?: IAccount | IAccountWalletProvider,
   ) {
     this.publicApiClient = new PublicApiClient(clientConfig);
     this.privateApiClient = new PrivateApiClient(clientConfig);
-    this.walletClient = new WalletClient(
-      clientConfig,
-      this.publicApiClient,
-      baseAccount,
-    );
+    // From: https://stackoverflow.com/a/73162923/10419533
```

Comment:
Not sure it's linked to what you are doing.
Here your simply defining the base account based on the wanted context. If you don't want to go the direction I suggested above, you could improve readibility by doing something like:
``` typescript
this.walletClient = new WalletClient(
        clientConfig,
        this.publicApiClient,
        provider(baseAccount, this.publicApiClient),
      );
 ```
 And somewhere else define the provider function:
 ```typescript
  private provider(): Account {
  if (!baseAccount) {
    return undefined;
  }

  if (isWeb3Account(baseAccount)) {
    return new Web3Account(baseAccount, this.publicApiClient);
  }
  
  return new WalletProviderAccount(baseAccount);
  }

Reply:
Changed the behavior to have IBaseAccount